packages/graphql/src/schema/resolvers/root-connection.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright (c) "Neo4j"
+ * Neo4j Sweden AB [http://neo4j.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { GraphQLResolveInfo } from "graphql";
+import { InputTypeComposer, SchemaComposer } from "graphql-compose";
+import { PageInfo } from "graphql-relay";
+import { execute } from "../../utils";
+import { translateRead } from "../../translate";
+import { Node } from "../../classes";
+import { Context } from "../../types";
+import getNeo4jResolveTree from "../../utils/get-neo4j-resolve-tree";
+import { isNeoInt } from "../../utils/utils";
+import { createConnectionWithEdgeProperties } from "../pagination";
+
+export default function rootConnectionResolver({ node, composer }: { node: Node; composer: SchemaComposer }) {
+    async function resolve(_root: any, _args: any, _context: unknown, info: GraphQLResolveInfo) {
+        const context = _context as Context;
+        const resolveTree = getNeo4jResolveTree(info);
+
+        const edgeTree = resolveTree.fieldsByTypeName[`${node.name}Connection`].edges;
+        const nodeTree = edgeTree.fieldsByTypeName[`${node.name}Edge`].node;
+
+        context.resolveTree = { ...nodeTree, args: resolveTree.args };
+        context.isRootConnectionField = true;
+
+        const [cypher, params] = translateRead({ context, node });
+
+        const executeResult = await execute({
+            cypher,
+            params,
+            defaultAccessMode: "READ",
+            context,
+        });
+
+        // TODO: Ask neo4j team how to resolve this issue
+        // TCK Tests Run the Resolver But Expect a Return Value
+        // So in order to get the tests to pass we have to return
+        // an empty connection. But it may be more preferable to
+        // throw an error if there is no record, but that might
+        // require modification to the "translateQuery" function
```

Comment:
I don't think an error should be thrown on no records found - errors should be reserved for actually unexpected behaviour, when no records is perfectly normal behaviour for this library. So this solution feels good! I believe this is also inline with the Relay specification.